readme/config.md
Almost all settings related to the graph being rendered are modify-able using the config object which is one of the parameters used in building your graph.

The configuration itself is split into different groups based on which part of the graph they affect. Here is a overview of each of them:

## graph

|Property		|Value Type		|Default|		Description|
|---------------|:-------------:|:-----:|-------------------|
|**palette**    |```string```	|*'Plain'*|Defines the color palette used while rendering the graph. Color palettes are predefined, ability to add custom color palettes is coming soon.|
|**background** |```#color```	| *#fff*|Defines the background color of the graph|
|**orientation**|```string```|*'Horizontal'*|Defines the orientation of the graph rendered.|

#### palette (```string```)
Defines the color palette used while rendering the graph. Color palettes are predefined, ability to add custom color palettes is coming soon.
  
Currently available palettes: Plain, Simple, RGB, Olive, Soil and Sky, Candid, Sulphide, New Moon, Nature, Sea, Earth, Lemon, Water, Grass, Hash, Soft, Brink, Bright, Lint

#### background (```#color```)
Defines the background color of the graph

#### orientation (```string```)
Defines the orientation of the graph rendered.
      
Possible values: Horizontal, Vertical

## meta

|Property		|Value Type		|Default|		Description|
|---------------|:-------------:|:-----:|-------------------|
|**position**	|```string```	|*'uv-div'*| CSS selector string of the dom element under which the graph needs to be rendered|
|**caption**	|```string```	|*''*	| Caption for the graph |
|**subCaption** |```string```	|*''*	| Subcaption for the graph. Its place right under the caption |
|**hlabel**		|```string```	|*''*	| Label for the horizontal axis to express what is being shown in that scale in your own terms|
|**vlabel**		|```string```	|*''*	| Label for the vertical axis to express what is being shown in that scale in your own terms|
|**hsublabel**	|```string```	|*''*	| Sublabel for the horizontal axis to express what is being shown in that scale in your own terms|
|**vsublabel**	|```string```	|*''*	| Sublabel for the vertical axis to express what is being shown in that scale in your own terms|

#### position (```string```)
CSS selector string of the dom element under which the graph needs to be rendered

#### caption (```string```)
Caption for the graph

#### subCaption (```string```)
Subcaption for the graph. Its placed right under the caption.

#### hlabel (```string```)
Label for the horizontal axis to express what is being shown in that scale in your own terms

#### vlabel (```string```)
Label for the vertical axis to express what is being shown in that scale in your own terms

#### hsublabel (```string```)
Sublabel for the horizontal axis to express anything else apart from the label in minor detail

#### vsublabel (```string```)
Sublabel for the vertical axis to express anything else apart from the label in minor detail

## dimension

|Property		|Value Type		|Default|		Description|
|---------------|:-------------:|:-----:|-------------------|
|**width**		|```string```/```number```|400|Width of the graph in pixels (without 'px')|
|**height**		|```string```/```number```|400|Height of the graph in pixels (without 'px')|


#### width (```string```/```number```)
Width of the graph in pixels (without 'px')

#### height (```string```/```value```)
Height of the graph in pixels (without 'px')

## margin

|Property		|Value Type		|Default|		Description|
|---------------|:-------------:|:-----:|-------------------|
|**top**		|```string```/```number```|50|Top margin of the graph in pixels (usually this space is occupied by Caption and Subcaption)|
|**bottom**		|```string```/```number```|150|Bottom Margin of the graph in pixels (usually this space is occupied by the horizontal axis label and caption)|
|**left**		|```string```/```number```|100|Left margin of the graph in pixels (usually this space is occupied by vertical axis label and caption)|
|**right**		|```string```/```number```|100|Right margin of the graph in pixels (usually this space is occupied by the legend)|


#### top (```string```/```value```)
Top margin of the graph in pixels (usually this space is occupied by Caption and Subcaption)

#### bottom (```string```/```value```)
Bottom Margin of the graph in pixels (usually this space is occupied by the horizontal axis label and caption)

#### left (```string```/```value```)
Left margin of the graph in pixels (usually this space is occupied by vertical axis label and caption)

#### right (```string```/```value```)
Right margin of the graph in pixels (usually this space is occupied by the legend)

## frame

|Property		|Value Type		|Default|		Description|
|---------------|:-------------:|:-----:|-------------------|
|**bgcolor**	|```#color```	|'#FFF'	|Defines the background color of the frame, includes the graph and the margins|

#### bgcolor (```#color```)
Defines the background color of the frame, includes the graph and the margins

## axis

|Property		|Value Type		|Default|		Description|
|---------------|:-------------:|:-----:|-------------------|
|**ticks**		|```number```	| 8		|Defines the number of ticks against the measuring axis|
|**subticks**	|```number```	| 2		| Defines the number of subticks against the measuring axis between ticks|
|**padding**	|```number```	| 5		| Defines the space between the axis and axis labels in pixels |
|**minor**		| ```number```	| -10	| Defines the minor tick length on the axis containing the label headers |
|**strokecolor**|```#color```	|'#000' | Defines the color used for axis strokes |
|**fontfamily**|```string```	|'Arial'| Defines the font family used for text in axis elements |
|**fontsize**|```number```| 14 | Defines the font size used for axis elements |
|**fontweight**| ```string``` | 'bold' | Defines the font weight used for text in axis elements |

#### ticks (```number```)
Defines the number of ticks against the measuring axis

#### subticks (```number```)
Defines the number of subticks against the measuring axis between ticks

#### padding (```number```)
Defines the space between between the axis and axis labels in pixels

#### minor (```number```)
Defines the minor tick length on the axis containing the label headers

#### strokecolor (```#color```)
Defines the color used for axis strokes

#### fontfamily (```string```)
Defines the font family used for text in axis elements

#### fontsize (```number```)
Defines the font size used for axis elements

#### fontweight (```string```)
Defines the font weight used for text in axis elements

## label

|Property		|Value Type		|Default|		Description|
|---------------|:-------------:|:-----:|-------------------|
|**fontfamily**	|```string```| 'Arial' | Font family used to show text in label elements |
|**fontsize**	|```number```|11| Font size used to show text in label elements|
|**fontweight**	|```string```| 'normal'| Font weight used to show text in label elements|
|**strokecolor**|```#color```|'#000'| Stroke color to show text in label elements|

#### fontfamily (```string```)
Font family used to show text in label elements

#### fontsize (```number```)
 Font size used to show text in label elements
 
#### fontweight (```string```)
Font weight used to show text in label elements

#### strokecolor (```#color```)
Stroke color to show text in label elements


 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 

src/util/config.js
uv.config = {
	graph : {
		palette : 'Brink',
		background : 'white',
		orientation : 'Horizontal',
		max : 0
	},

	meta : {
		position : '.uv_div',
		caption : 'Usage of browsers by the years',
		subcaption : 'Among major vendors',
		hlabel : 'Horizontal Axis Label',
		vlabel : 'Vertical Axis Label',
		hsublabel : 'h sublabel',
		vsublabel : 'v sublabel'
	},

	dimension : {
		width : 400,
		height : 400
	},

	margin : {
		top : 50,
		bottom : 150,
		left : 100,
		right : 100
	},

	frame : {
		bgcolor : 'white'
	},

	axis : {
		ticks : 8,
		subticks : 2,
		padding : 5,
		minor : -10,
		strokecolor : '#000',
		fontfamily : 'Arial',
		fontsize : '14',
		fontweight : 'bold'
	},

	label : {
		fontfamily : 'Arial',
		fontsize : '11',
		fontweight : 'normal',
		strokecolor : '#000'
	},

	scale : {
		type : 'linear',
		ordinality : 0.2
	},

	bar : {
		strokecolor : 'none',
		fontfamily : 'Arial',
		fontsize : '10',
		fontweight : 'bold',
		textcolor : '#000'
	},

	line : {
		interpolation : 'linear'
	},

	area : {
		interpolation : 'cardinal',
		offset : 'zero',
		opacity : 0.2
	},

	pie : {
		fontfamily : 'Arial',
		fontsize : '14',
		fontweight : 'normal',
		fontvariant : 'small-caps',
		fontfill : 'white',
		strokecolor : 'none',
		strokewidth : 2
	},
	
	donut : {
		fontfamily : 'Arial',
		fontsize : '14',
		fontweight : 'normal',
		fontvariant : 'small-caps',
		fontfill : 'white',
		factor : 0.4,
		strokecolor : 'none',
		strokewidth : 2
	},
	
	caption : {
		fontfamily : 'Arial',
		fontsize : '14',
		fontweight : 'bold',
		fontvariant : 'small-caps',
		textdecoration : 'none',
		hovercolor : 'dimgrey',
		textanchor : 'middle'
	},

	subCaption : {
		fontfamily : 'Arial',
		fontsize : '9',
		fontweight : 'normal',
		fontvariant : 'normal',
		textdecoration : 'none',
		textanchor : 'middle'
	},

	legend : {
		position : 'bottom',
		fontfamily : 'Arial',
		fontsize : '11',
		fontweight : 'normal',
		textmargin : 15,
		symbolsize : 10,
		inactive_color : '#DDD',
		legendstart : 0,
	},

	effects : {
		hovercolor : 'red',
		strokecolor : 'none',
		textcolor : 'black',
		duration : 800,
		hover : 400
	}
};